File: backend/src/main/java/cn/master/nexus/modules/system/entity/Project.java
package cn.master.nexus.modules.system.entity;

import cn.master.nexus.common.validation.groups.Created;
import cn.master.nexus.common.validation.groups.Updated;
import com.mybatisflex.annotation.Column;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.Table;
import com.mybatisflex.core.handler.JacksonTypeHandler;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 项目 实体类。
 *
 * @author 11's papa
 * @since 2026-03-03
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "项目")
@Table("project")
public class Project implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 项目ID
     */
    @Id
    @Schema(description = "项目ID", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "{project.id.not_blank}", groups = {Updated.class})
    @Size(min = 1, max = 50, message = "{project.id.length_range}", groups = {Created.class, Updated.class})
    private String id;

    /**
     * 项目编号
     */
    @Schema(description = "项目编号")
    private Long num;

    @Schema(description = "组织ID", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "{project.organization_id.not_blank}", groups = {Created.class})
    @Size(min = 1, max = 50, message = "{project.organization_id.length_range}", groups = {Created.class, Updated.class})
    private String organizationId;

    @Schema(description = "项目名称", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "{project.name.not_blank}", groups = {Created.class})
    @Size(min = 1, max = 255, message = "{project.name.length_range}", groups = {Created.class, Updated.class})
    private String name;

    /**
     * 是否启用
     */
    @Schema(description = "是否启用")
    private Boolean enable;

    /**
     * 模块设置
     */
    @Schema(description = "模块设置")
    @Column(typeHandler = JacksonTypeHandler.class)
    private List<String> moduleSetting;

    /**
     * 项目描述
     */
    @Schema(description = "项目描述")
    private String description;

    /**
     * 创建时间
     */
    @Column(onInsertValue = "now()")
    @Schema(description = "创建时间")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @Column(onInsertValue = "now()", onUpdateValue = "now()")
    @Schema(description = "更新时间")
    private LocalDateTime updateTime;

    /**
     * 修改人
     */
    @Schema(description = "修改人")
    private String updateUser;

    /**
     * 创建人
     */
    @Schema(description = "创建人")
    private String createUser;

    /**
     * 删除时间
     */
    @Schema(description = "删除时间")
    private LocalDateTime deleteTime;

    /**
     * 是否删除
     */
    @Schema(description = "是否删除")
    private Boolean deleted;

    /**
     * 删除人
     */
    @Schema(description = "删除人")
    private String deleteUser;

    /**
     * 全部资源池
     */
    @Schema(description = "全部资源池")
    private Boolean allResourcePool;

}

File: frontend/src/api/types/project.ts
export interface CreateOrUpdateSystemProjectParams {
    id?: string;
    // 项目名称
    name: string;
    num: string;
    // 项目描述
    description: string;
    // 启用或禁用
    enable: boolean;
    // 项目成员
    userIds: string[];
    // 模块配置
    moduleIds?: string[];
    // 所属组织
    organizationId?: string;
    // 资源池
    resourcePoolIds: string[];
    // 列表里的
    allResourcePool: boolean; // 默认全部资源池
}

export interface CreateOrUpdateOrgProjectParams {
    id?: string;
    name: string;
    description?: string;
    enable?: boolean;
    userIds?: string[];
    organizationId?: string;
    resourcePoolIds?: string[];
}
export interface OrgProjectTableItem {
    id: string;
    name: string;
    description: string;
    enable: boolean;
    adminList: UserExtend[];
    organizationId: string;
    organizationName: string;
    num: string;
    updateTime: number;
    createTime: number;
    memberCount: number;
    userIds: string[];
    resourcePoolIds: string[];
    orgAdmins: Record<string, any>;
    moduleIds: string[];
    // resourcePoolList: ResourcePoolItem[];
    allResourcePool: boolean;
}
export interface ProjectListItem {
    id: string;
    num: string;
    organizationId: string;
    name: string;
    description: string;
    createTime: number;
    updateTime: number;
    updateUser: string;
    createUser: string;
    deleteTime: number;
    deleted: boolean;
    deleteUser: string;
    enable: boolean;
    moduleIds: string[];
    moduleSetting: string;
}

export interface UserExtend {
    userId:string
    userName:string
}